### ISSUE TITLE
A guide for "migration to distroless" will be very helpful

### ISSUE THREAD

**Issue by aggaa-netapp** [GoogleContainerTools/distroless#908]
I have recently started to look into migrating our existing alpine OS based Java services to distroless images. And one thing I am missing is some kind of document to help/guide with this migration, which could cover some hard limitations of distroless images as well as some design patterns to solve common problems. 

For example
Problem: how do I pass JVM arguments
Solution: use JAVA_TOOL_OPTIONS

Problem: what if my JVM argument comes from runtime pod env variable VAR
Solution: use $(VAR) while building JAVA_TOOL_OPTIONS

Problem: what if I need to create a pod specific folder, before starting JVM, for some logs, i.e. with POD_NAME in the path ?
Solution: I believe this can't be done with distroless and is a hard limitation (I could be wrong)

Problem: I want to pass "-XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/some/host/mapped/path/heapdump_$(POD_NAME).hprof" to JVM when another pod env variable DUMP_HEAP_ON_OOM is “true”
Solution: I don’t know yet

**Comment by loosebazooka** [comment]
> Problem: what if I need to create a pod specific folder for some logs, i.e. with POD_NAME in the path ?
Solution: I believe this can't be done with distroless and is a hard limitation (I could be wrong)

If you want to dynamically create directories, does an init container not work? 
From: https://kubernetes.io/docs/concepts/workloads/pods/init-containers/
```
Init containers can contain utilities or custom code for setup that are not present in an app image. For example, there is no need to make an image FROM another image just to use a tool like sed, awk, python, or dig during setup.
```

**Comment by ajay-aggarwal** [comment]
Yes, it init containers will work, but in some way it defeats the purpose of using distroless images. We are exploring migrating to distroless images with the assumption that they are more secure (less attack surface). If we package an init container with shell access etc. alongside it, it defeats the overall purpose.

**Comment by loosebazooka** [comment]
How do you know the pod info? Is it known at build time?

**Comment by ajay-aggarwal** [comment]
Its runtime. Build time would have been easy (multistage)!

**Comment by loosebazooka** [comment]
I don't know that I understand your problem -- would be pretty difficult to create directories without any utilities to create them? 
- You could write a simple statically compiled program to do this for you (perhaps in go) and put that in your init container?

**Comment by ajay-aggarwal** [comment]
Sorry for the delay. Was distracted by log4J for last couple of days :) 
Yes, another distroless init container will absolutely work. I was only thinking in terms of "bash/mkdir" earlier.
